Source/Scene/processPbrMaterials.js

```diff
@@ -0,0 +1,786 @@
+define([
+        './ModelUtility',
+        '../Core/defined',
+        '../Core/defaultValue',
+        '../Core/WebGLConstants',
+        '../Core/webGLConstantToGlslType',
+        '../ThirdParty/GltfPipeline/addToArray',
+        '../ThirdParty/GltfPipeline/ForEach',
+        '../ThirdParty/GltfPipeline/hasExtension',
+        '../ThirdParty/GltfPipeline/numberOfComponentsForType'
+    ], function(
+        ModelUtility,
+        defined,
+        defaultValue,
+        WebGLConstants,
+        webGLConstantToGlslType,
+        addToArray,
+        ForEach,
+        hasExtension,
+        numberOfComponentsForType) {
+    'use strict';
+
+    /**
+     * @private
+     */
+    function processPbrMaterials(gltf, options) {
+        options = defaultValue(options, {});
+
+        // No need to create new techniques if they already exist,
+        // the shader should handle these values
+        if (hasExtension(gltf, 'KHR_techniques_webgl')) {
+            return gltf;
+        }
+
+        // All materials in glTF are PBR by default,
+        // so we should apply PBR unless no materials are found.
+        if (gltf.materials.length === 0) {
+            return gltf;
+        }
```

Comment:
Hit a weird edge case where a gltf didn't have any meshes so it didn't even create any default materials and failed at this line. I guess a quick check that `gltf.materials` is defined should fix this. CC @OmarShehata.